tests/tasks.spec.ts

```diff
@@ -29,44 +30,54 @@ test.describe.serial(
       await createTaskModalPage.clickCreateTaskButton();
 
       await projectMainView.assertTaskIsVisible(taskName);
+      await ApiHooks.deleteTask(request, taskName);
     });
 
-    test("Change task status to in progress", async ({ page }) => {
-      await page.goto("/");
+    test.describe("Editing existing tasks", () => {
+      test.beforeEach(async ({ page, request }) => {
+        await ApiHooks.createNewTask(request, taskName);
+        await page.goto("/");
+      });
 
-      const projectMainView = new ProjectMainView(page);
-      const editTaskModal = new EditTaskModal(page);
+      test.afterEach(async ({ request }) => {
+        await ApiHooks.deleteTask(request, taskName);
+      });
 
-      await projectMainView.openTaskModal(taskName);
-      await editTaskModal.changeTaskStatusToInProgress();
-      await editTaskModal.close();
+      test("Change task status to in progress", async ({ page }) => {
+        const projectMainView = new ProjectMainView(page);
+        const editTaskModal = new EditTaskModal(page);
 
-      await projectMainView.assertTaskIsInProgress(taskName);
-    });
+        await projectMainView.openTaskModal(taskName);
+        await editTaskModal.changeTaskStatusToInProgress();
+        await editTaskModal.close();
 
-    test("Change task name", async ({ page }) => {
-      await page.goto("/");
+        await projectMainView.assertTaskIsInProgress(taskName);
+      });
 
-      const projectMainView = new ProjectMainView(page);
-      const editTaskModal = new EditTaskModal(page);
+      test("Change task name", async ({ page }) => {
+        const projectMainView = new ProjectMainView(page);
+        const editTaskModal = new EditTaskModal(page);
 
-      await projectMainView.openTaskModal(taskName);
-      await editTaskModal.changeTaskName(changedTaskName);
-      await editTaskModal.close();
+        await projectMainView.openTaskModal(taskName);
+        taskName = changedTaskName;
```

Comment:
Nie wiem, czy jest potrzeba przepisania nowej nazwy do zmiennej z poprzednią nazwą; niżej "nowa nazwa" jest wykorzystywana tylko dwa razy i imo czytelniej jest, jak do metody do zmiany nazwy podajemy zmienną/stałą, której nazwa wskazuje, że nazwa jest inna

Reply:
Można zrobić bez tej zmiany, ale wtedy w hookach trzeba dać usuwanie tasków z nazwami taskName oraz changedTaskName, co podwaja ilość requestów i nie wiem czy się nie wysypie wtedy hook

Reply:
Hmm okej. Ale jak coś, to dwa hooki (requesty) jeden po drugim powinny zadziałać, a przynajmniej ja tak robię w teście duplikowania space - usuwam wcześniejszy i zduplikowany od razu i działa. Też, w metodzie `deleteTask` jest w zasadzie tylko wysłanie requestu - jeżeli coś się wysypie, to nie zatrzyma to przebiegu testu, co najwyżej przyjdzie odpowiedź z kodem 400 czy innym